core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3681,7 +3682,7 @@ public void heartbeat() throws InterruptedException {
         } catch (UnavailableException e) {
           LOG.warn("Failed to persist file {}, will retry later: {}", uri, e.toString());
           remove = false;
-        } catch (alluxio.exception.status.ResourceExhaustedException e) {
+        } catch (ResourceExhaustedException e) {
```

Comment:
We will only wait after the for-loop, is that fine?

Reply:
@calvinjia I took another look at this. Most of my work focused on fixing this in replication and in the replication checker. Since that's where I saw this issue. However taking a closer look here it looks like that the same behavior occurs if a user tries to run many persists concurrently (possible w/ many clients). I didn't see this behavior before because I was testing out of the CLI which persists each file individually. Reproducing the issue here just isn't as straightforward.

To me it looks like the persistence checker could have the same issue. In this case the solution would be to add the return statement at the end of this catch block which would would prevent any further jobs from being scheduled for the `mQuietPeriodSeconds` and then resume afterwards, mimicking the behavior of the `ReplicationChecker`